### PR TITLE
[RG-93] Project settings

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
Allow user modify project settings

### Describe the technical details
When project created user can modify project settings:
* via context menu in the project hierarchy
* via menu Project->Project setting

 ### Which part of the code base require a change
 
 - [x] Library: newproject, fordagcore, projnavigator
 - [ ] Plug-in: <Specify the plugin name>
 - [ ] Engine
 - [ ] Documentation
 - [x] Regression tests
 - [ ] Continous Integration (CI) scripts

#### Testcase1:
* run any script from testcase folder.
* copy somewhere project file
* Open and close project settings
* Compare latest project file with the one that was copied.

#### Testcase2:
* Create project with all available options
* Open project settings.
* It is expected that project settings are the same as it was created.

#### Testcase3:
* create project with some settings
* Open project settings and close several times.
* Nothing should be changed

#### Testcase4:
* create project with some settings
* Open project settings
* Change something
* Close settings and open again
* All changes should be saved.
* Repeat steps with all possible settings

### Screenshots
image
image
image

### CODE REVIEW

**Comment by KochynVolodymyr** [comment]
Need update for compilation unit

File: src/NewProject/ProjectManager/project_manager.cpp [hunk 1]
```diff
@@ -1359,14 +1274,8 @@ int ProjectManager::deleteFileSet(const QString& strSetName) {
   int ret = 0;
   ProjectFileSet* proFileSet =
       Project::Instance()->getProjectFileset(strSetName);
-  if (proFileSet && PROJECT_FILE_TYPE_DS == proFileSet->getSetType() &&
```
Comment (skyler-rs) [inline src/NewProject/ProjectManager/project_manager.cpp:1362]: Are these entries no longer needed?

**Comment by ravic-rs** [comment]
@KochynVolodymyr we got conflicts here as expected.

File: src/NewProject/new_project_dialog.cpp [hunk 2]
```diff
@@ -117,31 +157,38 @@ void newProjectDialog::Reset() {
...
-                              m_proTypeForm->getProjectType());
-    m_sumForm->setDeviceInfo(m_devicePlanForm->getSelectedDevice());
-    m_sumForm->setSourceCount(m_addSrcForm->getFileData().count(),
-                              m_addConstrsForm->getFileData().count());
-  } else {
-    NextBtn->setEnabled(true);
-    ui->buttonBox->button(QDialogButtonBox::Ok)->setEnabled(false);
-  }
+  m_addSrcForm = new addSourceForm(this);
+  m_addSrcForm->SetTitle("Modify Design Files");
```
Comment (ravic-rs) [inline src/NewProject/new_project_dialog.cpp:169]:
```suggestion
  m_addSrcForm->SetTitle("Design Files");
```

Comment (ravic-rs) [reply src/NewProject/new_project_dialog.cpp:169]: Remove the "Modify " from tab and titles. It implicit that users will modify the project using settings option.

**Review by TarasSydor — APPROVED** [review]
looks good

File: src/NewProject/ProjectManager/project_manager.cpp (hunk 1 above)
Comment (KochynVolodymyr) [reply src/NewProject/ProjectManager/project_manager.cpp:1362]: yes, the logic was changed long time ago and this doesn't make sense anymore

File: src/NewProject/new_project_dialog.cpp (hunk 2 above)
Comment (KochynVolodymyr) [reply src/NewProject/new_project_dialog.cpp:169]: but this is literally modification. Are you sure?

Comment (ravic-rs) [reply src/NewProject/new_project_dialog.cpp:169]: Yes, we don't need to explicitly mention it as opening up the settings implies that they are editable/modifiable.

Comment (KochynVolodymyr) [reply src/NewProject/new_project_dialog.cpp]: Done

**Pull request merged**